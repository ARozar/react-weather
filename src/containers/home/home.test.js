import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';
import { FiveDayForecastPage, styles } from './index';

describe('FiveDayForecastPage renders', () => {

  let props = { };

  beforeEach(()=>{
    props = {
      loading: false,
      error: false,
      title: 'Glasgow',
      days: [[], [], [], [], []],
      selectedDay:[],
      actions: {
        loadWeather: () => { },
        selectDay: () => { }
      },
      classes: styles
    };
  });

  const setup = (props) => shallow(<FiveDayForecastPage {...props} />);
  
  it('loading when loading', () => {

    props.loading = true;

    const wrapper = setup(props);

    expect(wrapper.find('h1').text()).toEqual('loading...');
  });

  it('Error when Error', () => {

    props.error = true;

    const wrapper = setup(props);

    expect(wrapper.find('h1').text()).toEqual('Error');
  });

  it('renders correctly when loaded with no errors', () => {
    //this is to support the material ui button element in the dayList component
    const options = {createNodeMock:(element) => {
      if (element.type === 'li') {
        return {
          focus() {},
        };
      }
      return null;
    }};
    
    const tree = renderer.create(<FiveDayForecastPage {...props} />, options).toJSON();

    expect(tree).toMatchSnapshot();
  });
});