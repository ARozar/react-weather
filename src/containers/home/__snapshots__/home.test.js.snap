// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiveDayForecastPage renders renders correctly when loaded with no errors 1`] = `
<div>
  <div
    className={
      Object {
        "flexGrow": 1,
        "margin": 30,
      }
    }
  >
    <div
      className="MuiGrid-typeContainer-1 MuiGrid-spacing-xs-24-17"
    >
      <div
        className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-31 MuiGrid-grid-sm-4-36"
      >
        <div>
          <div
            className="MuiPaper-root-84 MuiPaper-shadow2-88 MuiPaper-rounded-85"
          >
            <ul
              className="MuiList-root-111 MuiList-padding-112 MuiList-subheader-114"
            >
              <div
                className="MuiListSubheader-root-115"
              >
                <h2>
                  Glasgow
                </h2>
              </div>
              <li
                className="MuiButtonBase-root-128 MuiListItem-root-119 MuiListItem-gutters-126 MuiListItem-button-127 MuiListItem-default-122"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <div
                  className="MuiAvatar-root-130"
                />
                <div
                  className="MuiListItemText-root-133"
                >
                  
                </div>
                <span
                  className="MuiTouchRipple-root-138"
                />
              </li>
              <li
                className="MuiButtonBase-root-128 MuiListItem-root-119 MuiListItem-gutters-126 MuiListItem-button-127 MuiListItem-default-122"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <div
                  className="MuiAvatar-root-130"
                />
                <div
                  className="MuiListItemText-root-133"
                >
                  
                </div>
                <span
                  className="MuiTouchRipple-root-138"
                />
              </li>
              <li
                className="MuiButtonBase-root-128 MuiListItem-root-119 MuiListItem-gutters-126 MuiListItem-button-127 MuiListItem-default-122"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <div
                  className="MuiAvatar-root-130"
                />
                <div
                  className="MuiListItemText-root-133"
                >
                  
                </div>
                <span
                  className="MuiTouchRipple-root-138"
                />
              </li>
              <li
                className="MuiButtonBase-root-128 MuiListItem-root-119 MuiListItem-gutters-126 MuiListItem-button-127 MuiListItem-default-122"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <div
                  className="MuiAvatar-root-130"
                />
                <div
                  className="MuiListItemText-root-133"
                >
                  
                </div>
                <span
                  className="MuiTouchRipple-root-138"
                />
              </li>
              <li
                className="MuiButtonBase-root-128 MuiListItem-root-119 MuiListItem-gutters-126 MuiListItem-button-127 MuiListItem-default-122"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <div
                  className="MuiAvatar-root-130"
                />
                <div
                  className="MuiListItemText-root-133"
                >
                  
                </div>
                <span
                  className="MuiTouchRipple-root-138"
                />
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-31 MuiGrid-grid-sm-8-40"
      >
        <div>
          <div
            className="MuiPaper-root-84 MuiPaper-shadow2-88 MuiPaper-rounded-85"
          >
            <ul
              className="MuiList-root-111 MuiList-padding-112 MuiList-subheader-114"
            >
              <div
                className="MuiListSubheader-root-115"
              >
                <h3>
                  
                </h3>
              </div>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
